### ISSUE TITLE
Not working anymore for our Rails 4 app

### ISSUE THREAD

**Issue by ncri** [mattolson/heroku_rails_deflate#18]
It seems this gem has stopped working for our Rails app at some point. Assets are not compressed, I only see uncompressed load times with chrome:

![bildschirmfoto 2014-07-16 um 13 37 15](https://cloud.githubusercontent.com/assets/16925/3597643/424d268c-0cd5-11e4-8e64-08e1373aa5d7.png)

Here is the top part of our middleware stack:

```
use UTF8Cleaner::Middleware
use Rack::Sendfile
use ActionDispatch::Static
use Rack::Cache
use Rack::Deflater
use HerokuRailsDeflate::ServeZippedAssets
use #<ActiveSupport::Cache::Strategy::LocalCache::Middleware:0x007f520e8b62a0>
use Rack::Runtime
use Rack::MethodOverride
use ActionDispatch::RequestId
use Rails::Rack::Logger
use ActionDispatch::ShowExceptions
```

We have setup the config like in your readme.

Any idea what is going wrong? We use Rails 4.1.1.

**Comment by ncri** [comment]
Were are using rack-zippy now which works fine....

**Comment by mattolson** [comment]
Thanks for the bug report. To be honest, I haven't tested for Rails 4 yet,
but I'll check it out.

**Comment by ncri** [comment]
Okay, thx.

**Comment by laertiades** [comment]
I looks like you are serving your static assets before the gem gets a chance to deflate them.  You might try the following in production.rb, or application.rb:

```
  config.middleware.delete "ActionDispatch::Static"
  config.middleware.insert_before(Rack::Runtime, ActionDispatch::Static, paths["public"].first, config.static_cache_control)
```

**Comment by mcary** [comment]
@ncri, I agree with laertiades: HerokuRailsDeflate::ServeZippedAssets must appear before ActionDispatch::Static in the middleware stack.  And for my Rails 4 apps, it does -- works out of the box.

**Comment by ncri** [comment]
Well, you might be right, I haven't looked into this anymore, as we have another working solution now, as I wrote. 

If your idea works, that should be mentioned in the readme. Because out of the box it doesn't work for me.

**Comment by AlexEscalante** [comment]
I can report that I am using this gem on Heroku with Rails 4.2.1 and seems to work OK. I made a quick test using http://www.webpagetest.org/
